docs/internal-documentation/maintenance/ReleasePlan.md

```diff
@@ -17,13 +17,14 @@ the respective "release schedule"
 The minor or major features that are part of each release are determined by the development team.
 
 ## Release schedule
-The release schedule is to be created based on _minor releases_ of the Zonemaster parts, i.e. releases that changes then functionality but not the outward-facing API.
+The release schedule is to be created based on _minor releases_ or _major releases_ of the Zonemaster components, i.e.
+releases that changes functionality, and not only fixes bugs or erros.
 
-When needed, _minor releases_ are changed to _major releases_ of one or more parts, i.e. when functionality changes require changes to the outward-facing API.
+When needed, _patch releases_ of the components are published.
 
-When needed, _patch releases_ are published.
-
-When one or more parts are published as new release, the main Zonemaster repository is released listing the including components. At such event, there could be a mixture of major, minor and patch releases of the components.
+When one or more components have been updated (release has been published), the Zonemaster product is also
+released. Such a release of the Zonemaster product, could include a mixture of major, minor and patch releases of 
+the components.
```

Comment:
Bumping the product version for patch releases in between the scheduled minors and majors gives people an easy way to see that there is an update for one of the components. On the other hand it can mess with our planning if we're trying to have some set of issues ready for the 2018.4 release at the end of next year.

A possible solution is to let the product releases only specify the two first parts of component version numbers, and tell people to keep up with the latest patch release. This model also covers the situation I mentioned where we want to release a patch for the previous major because some serious issue came up and people haven't had time to upgrade.

Reply:
I think we should stick with the model that the version of the Zonemaster product is incremented each time any component is updated. We could decide to introduce a third level in the product versioning like Ubuntu, so if none of the components have been updated on major or minor level, then we increase a "patch" level of the product version, e.g. v2018.4 -> v2018.4.1.

I guess we could have a release of the Zonemaster product without any component being updated we want to release updates of documents or specifications. Such an update could be normal, e.g. v2018.4 -> v2018.5, or patch, e.g. v2018.4 -> v2018.4.1.